File: api-schedule/models/pair.js
const Link = require('./link')
class Pair{
    constructor(element) {
        this.id = element.id;
        this.dayofweekid = element.dayofweekid;
        this.teachersubjectid = element.teachersubjectid;
        this.audienceid = element.audienceid;
        this.datestart = element.datestart;
        this.numberweeks = element.numberweeks;
        this.groupid = element.groupid;
        this.typeofpairid = element.typeofpairid;
        this.timeid = element.timeid;
        this.links = [];
        this.links[0] = new Link('dayOfWeek', `${process.env.MAIN_URI}${process.env.MAIN_VERSION}${process.env.URI_DAY}${element.dayofweekid}`)
        this.links[1] = new Link('teachersubject', `${process.env.MAIN_URI}${process.env.MAIN_VERSION}${process.env.URI_TEACHERSUBJECT}${this.teachersubjectid}`)
        this.links[2] = new Link('audience', `${process.env.MAIN_URI}${process.env.MAIN_VERSION}${process.env.URI_AUDIENCE}${this.audienceid}`)
        this.links[3] = new Link('group', `${process.env.MAIN_URI}${process.env.MAIN_VERSION}${process.env.URI_GROUP}${this.groupid}`)
        this.links[4] = new Link('type', `${process.env.MAIN_URI}${process.env.MAIN_VERSION}${process.env.URI_TYPE}${this.typeofpairid}`)
        this.links[5] = new Link('time', `${process.env.MAIN_URI}${process.env.MAIN_VERSION}${process.env.URI_TIME}${this.timeid}`)

    }
}
module.exports = Pair;

File: desktop-client/MVVM/Models/Pair.cs
using System;

namespace MVVM.Models;

public class Pair : Model
{
    public DateOnly dateStart { get; set; }
    public DateOnly dateEnd { get; set; }
    public int? audienceId { get; set; }
    public int? teacherSubjectId { get; set; }
    public int? dayOfWeekId { get; set; }
    public int? timeId { get; set; }
    public int? typeOfPairId { get; set; }
    public int? groupId { get; set; }
    public Audience audience { get; set; }
    public TeacherSubject teacherSubject { get; set; }
    public DayOfWeek dayOfWeek { get; set; }
    public Time time { get; set; }   
    public TypeOfPair typeOfPair { get; set; }
    public Group group { get; set; }

    public Pair()
        : this(0, DateOnly.MinValue, DateOnly.MaxValue, 0, 0, 0, 0, 0, 0)
    {
        
    }

    public Pair(int id, DateOnly dateStart, DateOnly dateEnd, int audienceId, int teacherSubjectId, int dayOfWeekId,
        int timeId, int typeOfPairId, int groupId) 
        : base(id)
    {
        this.dateStart = dateStart;
        this.dateEnd = dateEnd;
        this.audienceId = audienceId;
        this.teacherSubjectId = teacherSubjectId;
        this.dayOfWeekId = dayOfWeekId;
        this.timeId = timeId;
        this.typeOfPairId = typeOfPairId;
        this.groupId = groupId;
    }
    
    public Pair(int id, DateOnly dateStart, DateOnly dateEnd, Audience audience, TeacherSubject teacherSubject, 
        DayOfWeek dayOfWeek, Time time, TypeOfPair typeOfPair, Group group)
        : this(id, dateStart, dateEnd, audience.id, teacherSubject.id, dayOfWeek.id, time.id, typeOfPair.id, group.id)
    {
        this.audience = audience;
        this.teacherSubject = teacherSubject;
        this.dayOfWeek = dayOfWeek;
        this.time = time;
        this.typeOfPair = typeOfPair;
        this.group = group;
    }
}